### PR TITLE
Various refactors in the way the UI is modeled

### PR DESCRIPTION
**Pull request by GabrielFleischer**
This PR's goal is to refactor the way the UI is modeled.

The main improvements are :
 - Remove the use of currentRollCall
 - Remove the use of currentElection
 - Refactor of `ElectionStartFragment` and to use RxJava
 - Refactor the `CastVoteFragment` to remove the multiple votes (for now) and simplify everything.
   Also make sure to subscribe to updates correctly
 - Correctly create the election and rollcall list

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java [hunk 1]
```diff
@@ -646,58 +640,10 @@ public LiveData<SingleEvent<String>> getScanWarningEvent() {
...
-    return currentRollCall;
-  }
-
-  public void setCurrentRollCall(RollCall rc) {
-    currentRollCall = rc;
-  }
-
   public void setCurrentRollCallId(String rollCallId) {
     currentRollCallId = rollCallId;
   }
```
Comment (Warhedgehog) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java:645]: Is this still necessary ?

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java [hunk 2]
```diff
@@ -773,34 +718,51 @@ public void subscribeToRollCalls(String laoId) {
...
+                                    "Could not fetch roll call with id " + id);
+                              }
+                            })
+                        .collect(Collectors.toList()))
+            .flatMap(
+                subjects ->
+                    Observable.combineLatest(
+                        subjects,
+                        rollCalls ->
+                            // Sort the election list. That way it stays somewhat consistent over
```
Comment (Warhedgehog) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java:740]: Roll call list ? 🙂

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Very impressive work. This is much better and clearer. A couple questions but very close to mergeable imo 👍

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java (hunk 2 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java:740]: ok

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/consensus/ElectionStartFragment.java [hunk 3]
```diff
@@ -35,16 +39,21 @@
...
 
   private final CompositeDisposable disposables = new CompositeDisposable();
+
   private ConsensusNode ownNode;
-  private Button electionStart;
-  private TextView electionStatus;
 
-  @Inject ElectionRepository electionRepository;
+  @Inject ElectionRepository electionRepo;
+  private com.github.dedis.popstellar.databinding.ElectionStartFragmentBinding binding;
```
[the remark this answers is not in the document]
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/consensus/ElectionStartFragment.java:55]: That's a good question 😄

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/election/fragments/CastVoteFragment.java [hunk 4]
```diff
@@ -36,125 +40,145 @@
...
-    // Setting the viewPager and its adapter
-    ViewPager2 viewPager2 = mCastVoteFragBinding.castVotePager;
-    CastVoteViewPagerAdapter adapter =
-        new CastVoteViewPagerAdapter(viewModel, mCastVoteFragBinding);
-    viewPager2.setAdapter(adapter);
-    viewPager2.setPageTransformer(new ZoomOutTransformer());
-    // Setting the indicator for horizontal swipe
-    CircleIndicator3 circleIndicator = mCastVoteFragBinding.swipeIndicator;
-    circleIndicator.setViewPager(viewPager2);
+  private boolean setLaoName(CastVoteFragmentBinding binding) {
```
[the remark this answers is not in the document]
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/election/fragments/CastVoteFragment.java:157]: I'm not sure to follow your remark 😅

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/election/fragments/CastVoteFragment.java [hunk 5]
```diff
@@ -36,125 +40,145 @@
...
+      return false;
+    } catch (UnknownLaoException e) {
+      logAndShow(requireContext(), TAG, R.string.error_no_lao);
+      return true;
+    }
+  }
 
-    voteButton.setOnClickListener(buttonListener);
-    return mCastVoteFragBinding.getRoot();
+  private boolean setElectionName(CastVoteFragmentBinding binding) {
```
[the remark this answers is not in the document]
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/election/fragments/CastVoteFragment.java:168]: Right, I'll do that

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java (hunk 1 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java:645]: Actually, it's still being used.

I'll create a commit to apply the same idea of "the fragment knows the rollcall id, not the viewmodel"

Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java:645]: Actually, it is used by the QRScanningFragment.

And as this is a big chunk of badly modularized code, I need to keep this for now.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/election/fragments/CastVoteFragment.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Warhedgehog) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/election/fragments/CastVoteFragment.java:157]: Forget it, I'm stupid 😆

**Review by Warhedgehog — APPROVED** [review]
LGTM !

**Pull request merged**